### ISSUE TITLE
Ability to adjust the spinner speed

### ISSUE THREAD

**Issue by musti-91** [davidhu2000/react-spinners#111]
It would be nice if we have control for speed of spinner, somehow that have used in my old project and we had to adjust it by CSS properties.

**Comment by wtesler** [comment]
Great feature request

**Comment by benji-meldcx** [comment]
Yes please. I can say there are spinners in this package that I love, but wouldn't use simply because the animation speed is not quite in beat with the rest of the app I'd intend it for.

Being able to optionally specify the duration of the animation for each spinner type would be amazing and imo the difference between some people using this library or not.

**Comment by davidhu2000** [comment]
It's been hard to find time to build new features. v0.10.0 is already a little big. I think this could be a great feature for v0.11.0

**Comment by benji-meldcx** [comment]
Thanks @davidhu2000, it's great to know this is on the Roadmap now. I may even be able to help with a PR. FYI, I noticed you've separated releases by project. Did you know you can tag issues with and filter by milestones?

**Comment by davidhu2000** [comment]
Starting to work on this feature now. How I'm thinking about implementing this is introducing a new prop called `speedMultiplier`. You'll pass in a number that adjusts the speed accordingly. Because many of the loaders include calculations for tying each piece together, allowing passing of an exact value might be tough in some cases. 

The downside of multiplier implementation would be the end user not knowing the actual speed (in seconds) of the animation. I'm wondering if that'll be an issue. 

For example, in BarLoader

```js
      animation: ${i === 1 ? long : short} 2.1s ${i === 2 ? "1.15s" : ""}
        ${i === 1
          ? "cubic-bezier(0.65, 0.815, 0.735, 0.395)"
          : "cubic-bezier(0.165, 0.84, 0.44, 1)"}
        infinite;
```

to simplify this for this example, let's pretend it's just 
```
animation: long 2.1s 1.15s cubic-bezier(...) infinite;
```

The speed is controlled by the 2nd value, so `2.1s`. 

If you pass in a prop like

```
<BarLoader speedMultiplier={2} />
```

The outputted animation would be 

```
animation: long 4.2s 1.15s cubic-bezier(...) infinite;
```

Thoughts?

**Comment by davidhu2000** [comment]
One complex one could be ClockLoader, where there are two different speeds of animation

```
      &:after {
        ...
        animation: ${rotate} 2s linear infinite;
      }

      &:before {
        ...
        animation: ${rotate} 8s linear infinite;
      }
```

I think having the end user pass in `2s` as an example, means it's hard to know how that 2s applies, is it applied to the after or the before?

**Comment by benji-meldcx** [comment]
Yeah given that the animations are looped infinitely and that some take significantly longer to return to their starting position than others, having an optional `speedMultiplier` defaulting to `1` makes sense.

**Comment by Beej126** [comment]
+want-this-too =) and am totally fine with a multiplier vs actual time input

**Comment by wtesler** [comment]
Yes, sounds like a solution. How is it going?

**Comment by musti-91** [comment]
Can we just make it easier as `speed` prop please.

**Comment by davidhu2000** [comment]
i don't think `speed` prop is explicit enough. I can easily see that interpreted as the exact timing of the animation vs just a multiplier.

**Comment by davidhu2000** [comment]
just published `0.11-alpha.3`, which includes `speedMultiplier` prop for `BarLoader`. please try it out and let me know what you think.

**Comment by Beej126** [comment]
neat! any chance you could put that on PulseLoader as well for testing?

**Comment by davidhu2000** [comment]
@Beej126  done. try it with `0.11-alpha.5`

**Comment by Beej126** [comment]
hey thanks... sorry to pull you along... i guess we're sortof accustomed to a slightly different fade approach to this... e.g. "three-bounce" from https://kyleamathews.github.io/react-spinkit/

**Comment by davidhu2000** [comment]
Ok. all speed multiplier props are published to `0.11.0-alpha.8`. 

You can also play with it on the demo site by adding a URL param

http://www.davidhu.io/react-spinners/?speed-multiplier=true

**Comment by benji-meldcx** [comment]
Amazing! Thank you so much for adding this feature @davidhu2000 ❤️

**Comment by davidhu2000** [comment]
This has been released as `v0.11`
